eoc_journal/eoc_journal.py

```diff
@@ -97,6 +106,52 @@ def student_view(self, context=None):
         )
         return fragment
 
+    def list_user_pb_answers_by_section(self):
+        """
+        Returns a list of dicts with pb-answers grouped by section.
+        """
+        # Get the selected blocks and their answers
+        blocks = self.list_pb_answers()
+        blocks = [b for b in blocks if b['id'] in self.selected_pb_answer_blocks]
+        course_id = self._get_course_id()
+        user_id = self._get_current_anonymous_user_id()
+        answers_names = [block['name'] for block in blocks]
+
+        answers = Answer.objects.filter(  # pylint: disable=no-member
+            course_key=course_id,
+            student_id=user_id,
+            name__in=answers_names
+        )
+
+        if answers.count() > 0:
+            # Map answer names to student inputs
+            students_inputs = {a.name: a.student_input for a in answers}
+
+            # Group answers by section
+            answers = OrderedDict()
+
+            for block in blocks:
+                name = block['name']
+                section = block['section']
+
+                if name in students_inputs:
```

Comment:
Oh, the reason for that is to avoid listing sections whose have not been answered yet.